Update to pass GNUPGHOME environmental variable as part of makepkg builds

gnfzdz opened the pull request:
Initialize GNUPGHOME from new config file option, build.GnupgHome. 

This PR should provide a minimal resolution for https://github.com/actionless/pikaur/issues/340

If the changes seem reasonable, I'll add test case(s). I'm also open to adding a related command line argument, but don't require it for my own use case.

actionless commented:
https://github.com/actionless/pikaur/issues/340#issuecomment-1230974077

actionless commented:
also flake is failing the style check: https://github.com/actionless/pikaur/runs/8081873696?check_suite_focus=true

gnfzdz commented:
Responded to #340 

I did notice the linting failure, I'll fix it tonight. I forgot to mention with the initial PR, but linting was initially failing locally complaining about a missing rule, which appears to have been related to [FS#75716](https://bugs.archlinux.org/task/75716). Sorry for the delay here.

actionless commented:
> Responded to https://github.com/actionless/pikaur/issues/340

in that case, instead of making it default value, please add a short FAQ entry about why this needed and usage examples (including those 'default' values)

> I did notice the linting failure, I'll fix it tonight. I forgot to mention with the initial PR, but linting was initially failing locally complaining about a missing rule, which appears to have been related to [FS#75716](https://bugs.archlinux.org/task/75716). Sorry for the delay here.

on master it's passing, you can rebase to get the latest code from there

actionless commented on README.md:
```diff
@@ -114,6 +114,10 @@ When doing sysupgrade ignore AUR packages which have `outofdate` mark.
 
 #### [build]
 
+##### GnupgHome (default: None)
```
you also need to update the doc accordingly

gnfzdz commented:
1. The push earlier was before noticing your message. Since it sounds like you're just okay with documenting, I reverted setting the default to `/etc/pacman.d/gnupg` and added a section to the FAQ.

2. The unit test `test_aur_rpc_didnt_fully_parsed_srcinfo_2` failed locally just now with a full regression, but the others that were failing earlier passed. I'll try to take a look again tomorrow.

3. I'm not sure if this is worth exposing as a command line argument, but the existing pacman argument (which pikaur inherited) is `--gpgdir`. Maybe it makes sense to change the configuration file option to `[build] gpgdir` as well and the hypothetical cli argument would be `--makepkg-gpgdir`?

actionless commented on pikaur/config.py:
```diff
@@ -105,7 +105,7 @@ def get_config_path() -> str:
         },
         'GnupgHome': {
             'type': 'str',
-            'default': ('/etc/pacman.d/gnupg/' if RUNNING_AS_ROOT else '')
```
i actually liked this idea - i meant to document this behavior in the readme

gnfzdz replied:
> i actually liked this idea - i meant to document this behavior in the readme

I've added it back

actionless commented:
> 3. I'm not sure if this is worth exposing as a command line argument, but the existing pacman argument (which pikaur inherited) is --gpgdir. Maybe it makes sense to change the configuration file option to [build] gpgdir as well and the hypothetical cli argument would be --makepkg-gpgdir?

up to you 😸

gnfzdz commented:
> > 3. I'm not sure if this is worth exposing as a command line argument, but the existing pacman argument (which pikaur inherited) is --gpgdir. Maybe it makes sense to change the configuration file option to [build] gpgdir as well and the hypothetical cli argument would be --makepkg-gpgdir?
> 
> up to you smile_cat

I've renamed the option as discussed above and added `--build-gpgdir` (for better consistency with the config file). I've also added a unit test that uses the option (and custom gpg home directory) to build a package requiring source validation (`zfs-utils`).

actionless commented on pikaur/args.py:
```diff
@@ -82,6 +82,7 @@ def get_pikaur_bool_opts() -> ArgSchema:
 
 def get_pikaur_str_opts() -> ArgSchema:
     return [
+        (None, 'build-gpgdir', None),
```
you need to add `PikaurConfig().build.GpgDir.get_str()` here instead of `build.py`

actionless commented:
nope, all good, i was just busy finishing mastering for the album, so havent been checking unrelated socials

https://share.amuse.io/album/actionlessloveless-loveless-trash

thanks a lot for your big effort on this feature
